### ISSUE TITLE
CSS / images not loading

### ISSUE THREAD

**Issue by taocrofl** [ccd0/4chan-x#1885]
Images/CSS stopped loading overnight. Running Chrome 66.0.3359.139. Here's what I see when I load the page. Interestingly, if I load the page with the developer console open, it renders fine. Then when I close it and hit refresh I get the problem again. Problem goes away when I disable 4chan-x, so I'm relatively confident that its the source of the problem. Ad-blocker (uBlock) is disabled so I don't think that's the problem either.

![image](https://user-images.githubusercontent.com/26388917/39628805-30e9a44e-4f78-11e8-9aef-f18916ca4e4b.png)

Console shows the following errors.

```
core.min.1014.js:1 Refused to load the script 'https://www.google-analytics.com/analytics.js' because it violates the following Content Security Policy directive: "script-src http://s.4cdn.org https://s.4cdn.org http://www.google.com https://www.google.com https://www.gstatic.com http://cdn.mathjax.org https://cdn.mathjax.org https://cdnjs.cloudflare.com 'self' 'unsafe-inline' 'unsafe-eval'".

(anonymous) @ core.min.1014.js:1
initAnalytics @ core.min.1014.js:1
contentLoaded @ core.min.1014.js:1

VM1103:3741 Refused to load the script 'blob:https://boards.4chan.org/8e0dc08f-af9e-4255-8685-0293cf3101e2' because it violates the following Content Security Policy directive: "script-src http://s.4cdn.org https://s.4cdn.org http://www.google.com https://www.google.com https://www.gstatic.com http://cdn.mathjax.org https://cdn.mathjax.org https://cdnjs.cloudflare.com 'self' 'unsafe-inline' 'unsafe-eval'".

t @ VM1103:3741
jc @ VM1103:3736
Ve.(anonymous function) @ VM1103:3335
(anonymous) @ VM1103:41
Qc.o.(anonymous function) @ VM1103:3997
(anonymous) @ VM1139 about:srcdoc:1

VM1103:194 Refused to load the script 'https://loadercdn.com/match.js' because it violates the following Content Security Policy directive: "script-src http://s.4cdn.org https://s.4cdn.org http://www.google.com https://www.google.com https://www.gstatic.com http://cdn.mathjax.org https://cdn.mathjax.org https://cdnjs.cloudflare.com 'self' 'unsafe-inline' 'unsafe-eval'".

Xn.(anonymous function) @ VM1103:194
p @ VM1103:312
v @ VM1103:309
it @ VM1103:307
(anonymous function) @ VM1103:3544
s @ VM1103:3839
kc @ VM1103:3828
r @ VM1103:3346
setTimeout (async)
o @ VM1103:3343
o @ VM1103:2622
Or @ VM1103:2620
Ve.(anonymous function) @ VM1103:3341
(anonymous) @ VM1103:41
Qc.o.(anonymous function) @ VM1103:3997
(anonymous) @ VM1139 about:srcdoc:1

```

**Comment by hihahihahoho** [comment]
Same problem here

**Comment by DBrickeen** [comment]
Same problem.  Occurs if I try to view the catalog, or if I open a thread and scroll to the bottom.

**Comment by guac24** [comment]
It's not ideal but for the moment if you enable ublock on the site 4chan x will work again

**Comment by taocrofl** [comment]
@guac24 i'm using ublock and it's broken for me. what are you doing exactly?

**Comment by Jaegrqualm** [comment]
Looks pretty familiar to how things were broken when the new advertisements went up. @DBrickeen 's account corresponds with the locations of the ads, too.

**Comment by guac24** [comment]
@taocrofl I am using ublock version 1.16.2 on an Opera browser
I have violent monkey 2.5.7 installed as a userscript too 
be sure to enable all in private mode as well

**Comment by ccd0** [comment]
>Ad-blocker (uBlock) is disabled

This is the problem. You need to enable it on 4chan.

**Comment by ccd0** [comment]
I updated the old workaround, but you should still use uBlock Origin.

**Comment by ccd0** [comment]
duplicates: #1886 #1887 #1888 #1889 #1890 #1891 #1892 #1893 #1894

**Comment by rld243** [comment]
@ccd0 I don't use any adblockers and the issue still persists after v1.13.15.11.

**Comment by Noah670** [comment]
@guac24 Enabling ublock origin does fix the problem but it also prevents image previews from loading. Hopefully this bug will be fixed soon.

**Comment by Hamuko** [comment]
uBlock Origin does pretty much nothing towards this issue. Still persists with 1.13.15.11, both with uBlock Origin enabled and disabled.

**Comment by Jaegrqualm** [comment]
Yeah, still breaks. My browser(qutebrowser) doesn't have ublock, but does have greasemonkey, so I'm kind of up a creek here.

**Comment by guac24** [comment]
![cap](https://user-images.githubusercontent.com/38983621/39665444-41b2bd1e-508c-11e8-9d90-24beafb40e0d.png)
@Noah670 Really? I'm getting the hover image previews just fine...

**Comment by taocrofl** [comment]
This issue resolved for me yesterday through no action of my own. I assume Hiro fixed something with the ads. Closing the issue.

**Comment by ccd0** [comment]
@Noah670
see https://github.com/ccd0/4chan-x/wiki/Frequently-Asked-Questions#error-reading-metadata

**Comment by ccd0** [comment]
Going to reopen if you don't mind until this is fixed for everyone. I have not noticed any changes in the ads.

**Comment by ccd0** [comment]
more dupes:
#1895 #1896 #1897 #1898 #1899 #1900

**Comment by WvTongeren** [comment]
Can confirm this is still happening.

I've tried enabling/disabling my adblock, that didn't work. Strangely enough the crash only appears to happen when you see the final comment in a thread, be that either in a thread itself or through the catalog or index. For example, I can safely browse through an entire thread of 300+ posts, but as soon as I get that final one on my screen it breaks.

Dunno what that means, but it might help with fixing the problem.

**Comment by DBrickeen** [comment]
The issue has been resolved for me with no changes other than updating 4chanX to 1.13.15.12.

**Comment by Jaegrqualm** [comment]
Yeah, works over here now, latest version.

**Comment by ccd0** [comment]
Some people still having problems: #1954

**Comment by ccd0** [comment]
`"Image Expansion" crashed` errors are caused by this.

Most reports have been for pre-1.13.15.12 versions: #1918 #1919 #1920 #1921 #1922 #1923 #1924 #1925 #1926 #1927 #1928 #1929 #1930 #1931 #1932 #1934 #1937 #1938 #1939 #1940 #1941 #1942 #1943 #1944 #1945 #1946 #1947 #1948 #1950 #1951 #1952

One report post-1.13.15.12: #1955

**Comment by ccd0** [comment]
I've noticed the people this issue so far post-1.14.0.15 are using Tampermonkey. It might be helpful to go to Tampermonkey's settings, change `General > Config mode` to `Advanced`, then scroll down to `Experimental > Inject Mode` and change it to `Instant`. Alternately you can try another userscript engine or the Chrome extension version.

**Comment by ubone** [comment]
#1955 is with violent monkey

uBO (with only uBOs filters) helped

**Comment by sinfex** [comment]
Well I changed inject mode to instant and now it works.

**Comment by Mudrock1** [comment]
Also wanted to echo that there are problems with violentmonkey. When switching to greasemonkey, there are no problems

**Comment by ccd0** [comment]
dups: #1953, #1957

**Comment by Confrade** [comment]
I installed the uBlock extentions and it fixed for a while, now every onow and then it happens again.

**Comment by Smithy3** [comment]
Strange, did Greasemonkey fix the issue for you?

**Comment by ccd0** [comment]
As of v1.14.1.2 4chan X disables the whitelist in the cases where it doesn't load fast enough to block everything. This seems to be common in Violentmonkey and sometimes happen in Tampermonkey without the `Inject Mode: Instant` option. In these cases nasty ad scripts may get through, but there's not much 4chan X can do about that. This will have to be fixed on the userscript engines' end.

dups pre-main fix: #1959 #1960 #1966
dups post-main fix, pre-v1.14.1.2: #1961 #1965

Closing this unless we start to see issues post-v1.14.1.2.

**Comment by ccd0** [comment]
more dups: #1970 #1973 #1975
